### PR TITLE
Fix Add Doctor Bugs

### PR DESCRIPTION
**Pull request by kimshitong**
Modified Delimiter

Overridden Hashcode

### CODE REVIEW

File: src/main/java/seedu/address/model/person/NameContainsKeywordsDoctorPredicate.java [hunk 1]
```diff
@@ -38,7 +39,10 @@ public boolean equals(Object other) {
                 (NameContainsKeywordsDoctorPredicate) other;
         return keywords.equals(otherNameContainsKeywordsPredicate.keywords);
     }
-
+    @Override
+    public int hashCode() {
+        return Objects.hash(keywords);
+    }
     @Override
```
Comment (yanghengtang) [inline src/main/java/seedu/address/model/person/NameContainsKeywordsDoctorPredicate.java:46]: Empty line before the override statement

**Review by yanghengtang — COMMENTED** [review]
Parameter delimiter bug still unfixed. Change from forward slash to back slash

File: src/main/java/seedu/address/logic/parser/CliSyntax.java
Comment (kimshitong) [inline src/main/java/seedu/address/logic/parser/CliSyntax.java:1]: @yanghengtang Hi yh, just confirming. will this be sufficient for delimitter?

Comment (yanghengtang) [reply src/main/java/seedu/address/logic/parser/CliSyntax.java:1]: Yes this is correct

File: src/main/java/seedu/address/model/person/NameContainsKeywordsDoctorPredicate.java (hunk 1 above)
Comment (kimshitong) [reply src/main/java/seedu/address/model/person/NameContainsKeywordsDoctorPredicate.java:48]: ok, just done

File: src/test/java/seedu/address/logic/commands/AddDoctorCommandTest.java [hunk 2]
```diff
@@ -88,6 +91,18 @@ public void toStringMethod() {
...
 
+    @Test
+    public void hashcode() {
+        AddDoctorCommand command = new AddDoctorCommand(ALICE);
+
+        // same value -> returns same hashcode
+        assertEquals(command.hashCode(), Objects.hash(ALICE));
+
+        // different value -> returns different hashcode
+        assertNotEquals(command.hashCode(), Objects.hash(BOB));
```
Comment (derekjxtan) [inline src/test/java/seedu/address/logic/commands/AddDoctorCommandTest.java:102]: actually different values giving the same hashcode doesnt violate lsp. The expected behaviour is a.equals(b) -> a.hashcode() == b.hashcode();

File: src/test/java/seedu/address/model/person/NameContainsKeywordsDoctorPredicateTest.java [hunk 3]
```diff
@@ -86,4 +88,20 @@ public void toStringMethod() {
...
+        List<String> keywords = List.of("keyword1", "keyword2");
+
+        // same value -> return same hashcode
+        NameContainsKeywordsDoctorPredicate command = new NameContainsKeywordsDoctorPredicate(keywords);
+        assertEquals(command.hashCode(), Objects.hash(keywords));
+
+        List<String> keywords2 = List.of("keyword1", "keyword3");
+
+        // different value -> returns different hashcode
+        assertNotEquals(command.hashCode(), Objects.hash(keywords2));
```
Comment (derekjxtan) [inline src/test/java/seedu/address/model/person/NameContainsKeywordsDoctorPredicateTest.java:103]: likewise for lsp

**Review by derekjxtan — COMMENTED** [review]
should add tests for the AddDoctorCommandParser class as well at some point.

**Review by derekjxtan — COMMENTED** [review]
lgtm

**Review by derekjxtan — APPROVED** [review]

**Pull request merged**